test/runtests.jl

```diff
@@ -1,66 +1,85 @@
+    pr = getproperties(d)
+    device_available = true
+end
 
-t = analog_input(d*"/ai0")
-@test typeof(getproperties(t)) == Dict{ASCIIString,Tuple{Any,Bool}}
-@test typeof(getproperties(t,d*"/ai0")) == Dict{ASCIIString,Tuple{Any,Bool}}
-@test typeof(t) == NIDAQ.AITask
-@test start(t) == nothing
-@test length(read(t, Float64, 3)) == 3
-@test stop(t) == nothing
-@test analog_input(t, d*"/ai1") == nothing
-@test start(t) == nothing
-@test length(read(t, UInt32, 6)) == 12
-@test stop(t) == nothing
-@test clear(t) == nothing
+if !(device_available && length(pr["AIPhysicalChans"]) == 0)
+    info("NIDAQ: measurement HW does not support any AIPhysicalChans, skipping part of the suite")
+else
+    t = analog_input(d*"/ai0")
+    @test typeof(getproperties(t)) == Dict{ASCIIString,Tuple{Any,Bool}}
+    @test typeof(getproperties(t,d*"/ai0")) == Dict{ASCIIString,Tuple{Any,Bool}}
+    @test typeof(t) == NIDAQ.AITask
+    @test start(t) == nothing
+    @test length(read(t, Float64, 3)) == 3
+    @test stop(t) == nothing
+    @test analog_input(t, d*"/ai1") == nothing
+    @test start(t) == nothing
+    @test length(read(t, UInt32, 6)) == 12
+    @test stop(t) == nothing
+    @test clear(t) == nothing
+end
 
-t = analog_output(d*"/ao0")
-@test typeof(t) == NIDAQ.AOTask
-@test start(t) == nothing
-@test write(t, rand(3)) == 3
-@test stop(t) == nothing
-@test analog_output(t, d*"/ao1") == nothing
-@test start(t) == nothing
-@test write(t, rand(UInt32,6,2)) == 6
-@test stop(t) == nothing
-@test clear(t) == nothing
+if !(device_available && length(pr["AOPhysicalChans"]) == 0)
+    info("NIDAQ: measurement HW does not support any AOPhysicalChans, skipping part of the suite")
+else
+    t = analog_output(d*"/ao0")
+    @test typeof(t) == NIDAQ.AOTask
+    @test start(t) == nothing
+    @test write(t, rand(3)) == 3
+    @test stop(t) == nothing
+    @test analog_output(t, d*"/ao1") == nothing
+    @test start(t) == nothing
+    @test write(t, rand(UInt32,6,2)) == 6
+    @test stop(t) == nothing
+    @test clear(t) == nothing
+end
 
-t = digital_input(d*"/Port0/Line0")
-@test typeof(t) == NIDAQ.DITask
-@test start(t) == nothing
-@test length(read(t, UInt32, 3)) == 3
-@test stop(t) == nothing
-@test digital_input(t, d*"/Port0/Line1") == nothing
-@test start(t) == nothing
-@test length(read(t, UInt8, 6)) == 12
-@test stop(t) == nothing
-@test clear(t) == nothing
+if !(device_available && length(pr["DILines"]) == 0)
+    info("NIDAQ: measurement HW does not support DILines, skipping part of the suite")
+else
+    t = digital_input(d*"/Port0/Line0")
+    @test typeof(t) == NIDAQ.DITask
+    @test start(t) == nothing
+    @test length(read(t, UInt32, 3)) == 3
+    @test stop(t) == nothing
+    @test digital_input(t, d*"/Port0/Line1") == nothing
+    @test start(t) == nothing
+    @test length(read(t, UInt8, 6)) == 12
+    @test stop(t) == nothing
+    @test clear(t) == nothing
 
-t = digital_output(d*"/Port0/Line0")
```

Comment:
some NIDAQ products have digital input but not digital output.  please add the `if device_available` logic around this section separately.